src/Nethermind/Nethermind.BeamWallet/Modules/Data/DataModule.cs

```diff
@@ -35,70 +35,131 @@ public class DataModule : IModule
         private readonly IEthJsonRpcClientProxy _ethJsonRpcClientProxy;
         private readonly Address _address;
         private readonly Timer _timer;
-        private readonly TimeSpan _interval;
         private decimal _balance;
         private Window _window;
         private Label _syncingInfoLabel;
         private Label _balanceValueLabel;
+        private CallTransactionModel _tokenTransaction;
+        private Label _tokenBalanceLabel;
+        private readonly IEnumerable<Token> _tokens = InitTokens();
+
         public event EventHandler<TransferClickedEventArgs> TransferClicked;
         
         public DataModule(IEthJsonRpcClientProxy ethJsonRpcClientProxy, string address)
         {
             _ethJsonRpcClientProxy = ethJsonRpcClientProxy;
             _address = new Address(address);
-            _interval = TimeSpan.FromSeconds(5);
-            _timer = new Timer(Update, null, TimeSpan.Zero, _interval);
+            TimeSpan interval = TimeSpan.FromSeconds(5);
+            _timer = new Timer(Update, null, TimeSpan.Zero, interval);
         }
 
-        public async Task<Window> InitAsync()
+        public Task<Window> InitAsync()
         {
             _window = new Window("Data")
             {
                 X = 0,
                 Y = 0,
-                Width = Dim.Fill(),
+                Width = Dim.Fill(),    
                 Height = Dim.Fill()
             };
             Application.Top.Add(_window);
-            await RenderBalanceAsync();
+            RenderBalanceAsync();
 
-            return _window;
+            return Task.FromResult(_window);
         }
 
         private void Update(object state)
         {
             _ = UpdateBalanceAsync();
+            // _ = UpdateTokenBalanceAsync();
         }
 
         private async Task UpdateBalanceAsync()
         {
-            if (_syncingInfoLabel is null || _balanceValueLabel is null)
+            if (_syncingInfoLabel is null || _balanceValueLabel is null || _address is null)
             {
                 return;
             }
-            
-            var balanceResult = await _ethJsonRpcClientProxy.eth_getBalance(_address);
-            
-            _balance = WeiToEth(balanceResult.Result.ToString());
+
+            var balance = await GetBalanceAsync();
+            if (!balance.HasValue)    
+            {
+                return;
+            }
+
+            _balance = balance.Value;
             _window.Remove(_balanceValueLabel);
-            _balanceValueLabel = new Label(70, 1, $"{_balance} ETH");
+            _balanceValueLabel = new Label(70, 1, $"{_balance} ETH (refreshing every 5s)");
```

Comment:
I know it is a small app but worth to read about MVC / MVVM / whatever the latest thing is on separating UI and data -> here we have a direct update of a control